webui/react/src/components/MultiSelect.tsx

```diff
@@ -0,0 +1,85 @@
+import { Select } from 'antd';
+import { SelectValue } from 'antd/es/select';
+import React, { useCallback, useMemo } from 'react';
+
+import SelectFilter from './SelectFilter';
+
+interface LabeledValue {
+  label: number|string;
+  value: number|string;
+}
+
+interface Props {
+  label: string;
+  onChange?: (value: (number|string)[]) => void;
+  options: (number|string|LabeledValue)[];
+  value?: (number|string)[];
+}
+
+const ALL_VALUE = 'All';
+
+const { Option } = Select;
+
+const MultiSelect: React.FC<Props> = ({ label, onChange, options, value }: Props) => {
+
+  const handleSelect = useCallback((option: SelectValue) => {
+    if (!onChange) return;
+
+    const optionString = option.toString();
+    if (optionString === ALL_VALUE) {
+      onChange([]);
+      if (document && document.activeElement) {
+        (document.activeElement as HTMLElement).blur();
+      }
+      return;
+    }
+
+    const newValue = Array.isArray(value) ? [ ...value ] : [];
+    if (newValue.indexOf(optionString) === -1) newValue.push(optionString);
+    onChange(newValue);
+  }, [ onChange, value ]);
+
+  const handleDeselect = useCallback((option: SelectValue) => {
+    if (!onChange) return;
+
+    const newValue = Array.isArray(value) ? [ ...value ] : [];
+    const optionString = option.toString();
+    const index = newValue.indexOf(optionString);
+    if (index !== -1) newValue.splice(index, 1);
+    onChange(newValue);
+  }, [ onChange, value ]);
+
+  const selectOptions = useMemo(() => {
+    return options.map((item: number|string|LabeledValue) => (
+      [ 'string', 'number' ].indexOf(typeof item) >= 0 ? (
```

Comment:
**nit:** `[ 'string', 'number' ].contains(typeof item)`

Reply:
I guess you meant `included` (which I didn't know). Fixed.

Reply:
er right `includes` 🤦